container.go

```diff
@@ -272,6 +280,98 @@ func (c *container) Update(ctx context.Context, opts ...UpdateContainerOpts) err
 	return nil
 }
 
+func (c *container) Checkpoint(ctx context.Context, ref string, opts ...CheckpointOpts) (Image, error) {
+	index := &ocispec.Index{
+		Versioned: ver.Versioned{
+			SchemaVersion: 2,
+		},
+		Annotations: make(map[string]string),
+	}
+	copts := &options.CheckpointOptions{
+		Exit:                false,
+		OpenTcp:             false,
+		ExternalUnixSockets: false,
+		Terminal:            false,
+		FileLocks:           true,
+		EmptyNamespaces:     nil,
+	}
+	img, err := c.Image(ctx)
+	if err != nil {
+		return nil, err
+	}
+	index.Annotations["image.name"] = img.Name()
+
+	ctx, done, err := c.client.WithLease(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer done(ctx)
+
+	// pause task to checkpoint
+	if err := func(ctx context.Context) error {
+		task, err := c.Task(ctx, nil)
+		if err != nil {
+			if errdefs.IsNotFound(err) {
+				return nil
+			}
+			return err
+		}
+		if err := task.Pause(ctx); err != nil {
+			return err
+		}
+		defer task.Resume(ctx)
+
+		info, err := c.Info(ctx)
+		if err != nil {
+			return err
+		}
+
+		// add runtime info to index
+		index.Annotations["runtime.name"] = info.Runtime.Name
```

Comment:
Can we make all of this Opts?

Reply:
I changed this to be a separate media type that can optionally be restored instead.